iaso/api/tasks/__init__.py

```diff
@@ -70,3 +79,211 @@ def get_queryset(self):
+
+    def create(self, request):
+        serializer_class = self.get_serializer_class()
+        serializer = serializer_class(data=request.data, context={"request": request})
+        serializer.is_valid(raise_exception=True)
+        task = serializer.save()
+        data = serializer.validated_data
+        config = data.get("config", None)
+        slug = data.get("slug", None)
+        id_field = data.get("id_field", None)
+        status = self.launch_task(slug=slug, config=config, id_field=id_field, task_id=task.pk)
+        task.status = status
+        task.save()
+        return Response({"task": TaskSerializer(instance=task).data})
+
+    # slug is the slug of the Config object
+    # config is the pipeline specific config (the args of the pipeline method grouped in a dict)
+    # task_id will be passed by the task decorator
+    # id_field is a field to filter from to find the relevant active run, eg: a country id for lqas refresh
+    def launch_task(self, slug, config={}, task_id=None, id_field=None):
+        try:
+            # The config Model should be moved to Iaso as well
+            pipeline_config = get_object_or_404(Config, slug=slug)
+            pipeline_version = pipeline_config.content["pipeline_version"]
+            pipeline = pipeline_config.content["pipeline"]
+            pipeline_target = pipeline_config.content["oh_pipeline_target"]
+            openhexa_url = pipeline_config.content["openhexa_url"]
+            openhexa_token = pipeline_config.content["openhexa_token"]
+        except:
+            logger.exception("Could not fetch openhexa config")
+            return ERRORED
+
+        id_field_key = ""
+        id_field_value = None
+
+        if id_field:
+            try:
+                id_field_key = list(id_field.keys())[0]
+                id_field_value = list(id_field.values())[0]
+            except:
+                logger.exception(f"Bad id_field configuration.Expected non-empty dict, got {id_field}")
+                return ERRORED
+
+        transport = RequestsHTTPTransport(
+            url=openhexa_url,
+            verify=True,
+            headers={"Authorization": f"Bearer {openhexa_token}"},
+        )
+        client = Client(transport=transport, fetch_schema_from_transport=True)
+        get_runs = gql(
+            """
+        query pipeline {
+            pipeline(id: "%s"){
+                runs{
+                    items{
+                        run_id
+                        status
+                        config
+                    }
+                }
+            }
+        }
+        """
+            % (pipeline)
+        )
+        try:
+            latest_runs = client.execute(get_runs)
+            # Warning the query will only return the last 10 runs
+            active_runs = [
+                run
+                for run in latest_runs["pipeline"]["runs"]["items"]
+                if (run["status"] not in ["queued", "success", "failed"])
```

Comment:
What about using constants instead of string for "queued" et al.

Reply:
It comes from OpenHexa and is only used there (Iaso's are uppercase), so I don't know if there's much to gain